s3/s3_transport/include/s3_transport.hpp

```diff
@@ -0,0 +1,1901 @@
+
+                    // read from cache
+
+                    write_callback.reset(new s3_upload::callback_for_write_from_cache_to_s3<buffer_type>
+                            (bucket_context_, upload_manager_));
+
+                    s3_upload::callback_for_write_from_cache_to_s3<buffer_type>
+                        *write_callback_from_cache =
+                        static_cast<s3_upload::callback_for_write_from_cache_to_s3<buffer_type>*>
+                        (write_callback.get());
+
+                    write_callback_from_cache->set_and_open_cache_file(cache_file_path_);
+
+                    write_callback->content_length = get_cache_file_size();
+                    write_callback->offset = 0;
+
+
+                } else {
+
+                    // Read from buffer
+
+                    write_callback.reset(new
+                            s3_upload::callback_for_write_from_buffer_to_s3<buffer_type>(
+                                bucket_context_, upload_manager_, circular_buffer_));
+
+                    s3_upload::callback_for_write_from_buffer_to_s3<buffer_type>
+                        *write_callback_from_buffer =
+                        static_cast<s3_upload::callback_for_write_from_buffer_to_s3<buffer_type>*>
+                        (write_callback.get());
+
+                    upload_page<buffer_type> page;
+
+                    // read the first page
+                    rodsLog(config_.debug_log_level, "%s:%d (%s) [[%lu]] waiting to read\n",
+                            __FILE__, __LINE__, __FUNCTION__, get_thread_identifier());
+
+                    circular_buffer_.pop_front(page);
+
+                    rodsLog(config_.debug_log_level, "%s:%d (%s) [[%lu]] read page [buffer=%p][buffer_size=%lu][terminate_flag=%d]\n",
+                            __FILE__, __LINE__, __FUNCTION__, get_thread_identifier(), page.buffer.data(),
+                            page.buffer.size(), page.terminate_flag);
+
+                    write_callback_from_buffer->buffer = page.buffer;
+                    write_callback->content_length = config_.object_size;
+
+                }
+
+                write_callback->offset = 0;
+                write_callback->enable_md5 = config_.enable_md5_flag;
+                write_callback->thread_identifier = get_thread_identifier();
+                write_callback->object_key = object_key_;
+                write_callback->shmem_key = shmem_key_;
+                write_callback->shared_memory_timeout_in_seconds = config_.shared_memory_timeout_in_seconds;
+
+                std::stringstream msg;
+
+                put_props_.md5 = nullptr;
+                //if ( partData.enable_md5 ) {
+                //    // jjames - not sure how to do MD5 piecewise
+                //    putProps->md5 = s3CalcMD5( partData.put_object_data.fd,
+                //    partData.put_object_data.offset,
+                //    partData.put_object_data.content_length, resource_name );
+                //}
```

Comment:
same here

Reply:
Same as before.  See issue #1947.  Leaving the enable_md5 flag in the code and will comment that it currently has no effect.